File: SecondYear/praveer7398_PraveerNandan_2226ce1011_2/Week_3/sorting_three_groups.java
import java.util.Scanner;

class sorting_three_groups {
    public int minimumOperations(int[] v) {
        int n = v.length;
        int[] dp = new int[n];

        for (int i = 0; i < n; i++) {
            dp[i] = 1;
        }

        int m = 1;

        for (int i = 1; i < n; i++) {
            for (int j = i - 1; j >= 0; j--) {
                if (v[i] >= v[j]) {
                    dp[i] = Math.max(dp[i], 1 + dp[j]);
                    m = Math.max(m, dp[i]);
                }
            }
        }

        return n - m;
    }

    public static void main(String[] args) {
        sorting_three_groups solution = new sorting_three_groups();
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter the size of the array: ");
        int n = scanner.nextInt();
        int[] v = new int[n];

        System.out.println("Enter the elements of the array:");
        for (int i = 0; i < n; i++) {
            v[i] = scanner.nextInt();
        }

        int minOps = solution.minimumOperations(v);
        System.out.println("Minimum Operations: " + minOps);

        scanner.close();
    }
}

File: SecondYear/ApoorvaSri123_ApoorvaSrivastava_2226it1116_2/Week_3/sorting_three_groups.c
#include <stdio.h>

int minimumOperations(int v[], int n) {
    int dp[1000]; // Assuming a maximum of 1000 elements
    for (int i = 0; i < n; i++) {
        dp[i] = 1;
    }

    int m = 1;

    for (int i = 1; i < n; i++) {
        for (int j = i - 1; j >= 0; j--) {
            if (v[i] >= v[j]) {
                dp[i] = (dp[i] > (1 + dp[j])) ? dp[i] : (1 + dp[j]);
                m = (m > dp[i]) ? m : dp[i];
            }
        }
    }

    return n - m;
}

int main() {
    int n;
    printf("Enter the size of the array: ");
    scanf("%d", &n);
    int v[1000]; // Assuming a maximum of 1000 elements

    printf("Enter the elements of the array:\n");
    for (int i = 0; i < n; i++) {
        scanf("%d", &v[i]);
    }

    int minOps = minimumOperations(v, n);
    printf("Minimum Operations: %d\n", minOps);

    return 0;
}
